#!/usr/bin/env python
PACKAGE = "ardronecontrol"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Kp_x", double_t, 0, "A double parameter",    .035, 0,   0.5)
gen.add("Kd_x", double_t, 0, "A double parameter",    .25, 0,   1)
gen.add("Ki_x", double_t, 0, "A double parameter",    0, 0,   0.002)
gen.add("set_x", double_t, 0, "A double parameter",    1, 0,   2)

gen.add("Kp_y", double_t, 0, "A double parameter",    .035, 0,   0.5)
gen.add("Kd_y", double_t, 0, "A double parameter",    .25, 0,   1)
gen.add("Ki_y", double_t, 0, "A double parameter",    0, 0,   0.002)
gen.add("set_y", double_t, 0, "A double parameter",    1, 0,   2)

gen.add("Kp_x_vel", double_t, 0, "A double parameter",    .4, 0,   2)
gen.add("Kd_x_vel", double_t, 0, "A double parameter",    .8, 0,   2)
gen.add("Ki_x_vel", double_t, 0, "A double parameter",    0, 0,   2)
gen.add("set_x_vel", double_t, 0, "A double parameter",    1, 0,   2)


gen.add("Kp_x_acc", double_t, 0, "A double parameter",    .4, 0,   2)
gen.add("Kd_x_acc", double_t, 0, "A double parameter",    .8, 0,   2)
gen.add("Ki_x_acc", double_t, 0, "A double parameter",    0, 0,   2)
gen.add("set_x_acc", double_t, 0, "A double parameter",    1, 0,   2)

exit(gen.generate(PACKAGE, "ardronecontrol", "PIDset"))
